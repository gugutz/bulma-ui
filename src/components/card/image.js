import React from 'react'
import PropTypes from 'prop-types'

const Image = ({children, ...props}) => (
  <div className="card-image" {...props}>
    {children}
  </div>
)

Image.propTypes = {
  children: PropTypes.node.isRequired
}

export default Image
