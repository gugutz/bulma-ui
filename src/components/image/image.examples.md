[Checkout Bulma documentation](https://bulma.io/documentation/elements/image/).

Because images can take a few seconds to load (or not at all), use the `Image` component to specify a precisely sized container or the ratio of the image so that your layout isn't broken because of image loading or image errors.

### Basic Usage

```jsx
<Image 
  size="128x128"
  alt="alternate text"
  src="https://bulma.io/images/placeholders/128x128.png" 
/>
```

### Images with fixed size

```jsx
<div style={{width:200}}>
  <Image
    size="16x16" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/16x16.png" 
  />

  <br />
  <Image
    size="24x24" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/24x24.png" 
  />

  <br />
  <Image
    size="48x48" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/48x48.png" 
  />

  <br />
  <Image
    size="64x64" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/64x64.png" 
  />

  <br />
  <Image
    size="96x96" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/96x96.png" 
  />

  <br />
  <Image
    size="128x128" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/128x128.png" 
  />
</div>
```

### Images with fixed ratio

```jsx
<div style={{width:200}}>
  <p>square</p>
  <Image
    size="square" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/empty.png" 
  />

  <br />
  <p>1by1</p>
  <Image
    size="1by1" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/empty.png" 
  />

  <br />
  <p>5by4</p>
  <Image
    size="5by4" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/empty.png" 
  />

  <br />
  <p>4by3</p>
  <Image
    size="4by3" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/empty.png" 
  />

  <br />
  <p>3by2</p>
  <Image
    size="3by2" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/empty.png" 
  />

  <br />
  <p>5by3</p>
  <Image
    size="5by3" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/empty.png" 
  />

  <br />
  <p>16by9</p>
  <Image
    size="16by9" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/empty.png" 
  />

  <br />
  <p>2by1</p>
  <Image
    size="2by1" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/empty.png" 
  />

  <br />
  <p>3by1</p>
  <Image
    size="3by1" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/empty.png" 
  />

  <br />
  <p>4by5</p>
  <Image
    size="4by5" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/empty.png" 
  />

  <br />
  <p>3by4</p>
  <Image
    size="3by4" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/empty.png" 
  />

  <br />
  <p>2by3</p>
  <Image
    size="2by3" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/empty.png" 
  />

  <br />
  <p>3by5</p>  
  <Image
    size="3by5" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/empty.png" 
  />

  <br />
  <p>9by16</p>  
  <Image
    size="9by16" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/empty.png" 
  />

  <br />
  <p>1by2</p>  
  <Image
    size="1by2" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/empty.png" 
  />

  <br />
  <p>1by3</p>  
  <Image
    size="1by3" 
    alt="alternate text"
    src="https://bulma.io/images/placeholders/empty.png" 
  />
</div>
```

