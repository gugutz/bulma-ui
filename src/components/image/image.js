import React from 'react'
import PropTypes from 'prop-types'
import {classnames} from '../../utils'

export const sizes = [
  '16x16',
  '24x24',
  '48x48',
  '64x64',
  '96x96',
  '128x128',
  'square',
  '1by1',
  '5by4',
  '4by3',
  '3by2',
  '5by3',
  '16by9',
  '2by1',
  '3by1',
  '4by5',
  '3by4',
  '2by3',
  '3by5',
  '9by16',
  '1by2',
  '1by3'
]

const Image = ({size, src, alt, ...props}) => (
  <figure className={classnames('image', {[`is-${size}`]: size !== null})}>
    <img src={src} alt={alt} {...props}/>
  </figure>
)

Image.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  size: PropTypes.oneOf(sizes)
}

Image.defaultProps = {
  size: null
}

export default Image
