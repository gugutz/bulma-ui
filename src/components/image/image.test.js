import React from 'react'
import renderer from 'react-test-renderer'
import {default as Image, sizes} from './image'

describe('<Image /> component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Image
          alt="something"
          src="https://bulma.io/images/placeholders/128x128.png"
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  for (const size of sizes) {
    it('should support size prop with ' + size, () => {
      const tree = renderer
        .create(
          <Image
            size={size}
            alt="something"
            src="https://bulma.io/images/placeholders/128x128.png"
          />
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }
})
