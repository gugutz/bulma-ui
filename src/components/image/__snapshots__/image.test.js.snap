// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> component should render correctly 1`] = `
<figure
  className="image"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 1by1 1`] = `
<figure
  className="image is-1by1"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 1by2 1`] = `
<figure
  className="image is-1by2"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 1by3 1`] = `
<figure
  className="image is-1by3"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 2by1 1`] = `
<figure
  className="image is-2by1"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 2by3 1`] = `
<figure
  className="image is-2by3"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 3by1 1`] = `
<figure
  className="image is-3by1"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 3by2 1`] = `
<figure
  className="image is-3by2"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 3by4 1`] = `
<figure
  className="image is-3by4"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 3by5 1`] = `
<figure
  className="image is-3by5"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 4by3 1`] = `
<figure
  className="image is-4by3"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 4by5 1`] = `
<figure
  className="image is-4by5"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 5by3 1`] = `
<figure
  className="image is-5by3"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 5by4 1`] = `
<figure
  className="image is-5by4"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 9by16 1`] = `
<figure
  className="image is-9by16"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 16by9 1`] = `
<figure
  className="image is-16by9"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 16x16 1`] = `
<figure
  className="image is-16x16"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 24x24 1`] = `
<figure
  className="image is-24x24"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 48x48 1`] = `
<figure
  className="image is-48x48"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 64x64 1`] = `
<figure
  className="image is-64x64"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 96x96 1`] = `
<figure
  className="image is-96x96"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with 128x128 1`] = `
<figure
  className="image is-128x128"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`<Image /> component should support size prop with square 1`] = `
<figure
  className="image is-square"
>
  <img
    alt="something"
    src="https://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;
